force-app/main/default/classes/StaticTask.cls

```diff
@@ -1,38 +1,94 @@
+      Map<Integer, List<String>> sObjectFieldsMap = StaticTask.splitStringListByObjectFieldMap(payloadAsStringList);
+      List<Contact> contactList = StaticTask.splitObjectFieldMapByRecords(sObjectFieldsMap);
+      Map<String, Contact> emailAndContactMap = StaticTask.createEmailAndContactMap(contactList);
+
+      return emailAndContactMap;
+  }
+
+
+  public static List<String> splitPayloadByElementStringList (String payload) {
+    return payload.split(';');
+  }
+
+  
+  public static Map<Integer, List<String>> splitStringListByObjectFieldMap (List<String> stringList) {
+    
+    Map<Integer, List<String>> sObjectFieldMap = new Map<Integer, List<String>>();
+    Integer i = 0;
+    
+    for (String element : stringList) {
+      sObjectFieldMap.put(i, element.split(','));
+      i++;
+    }
+
+    return sObjectFieldMap;
+  }
+
+
+  public static List<Contact> splitObjectFieldMapByRecords (Map<Integer, List<String>> integerStringListMap) {
+    
+    List<Contact> contactList = new List<Contact>();
+
+    for (Integer i = 0; i < integerStringListMap.size(); i++) {
+      Contact record = new Contact();
+      List<String> singleListFromMap = integerStringListMap.get(i);
+
+      for (String fieldString : singleListFromMap) {
+        if (fieldString.contains('FirstName') && StaticTask.isRightPartOfPairBlank(fieldString)) {
+          record.FirstName = StaticTask.getRightPartOfPairWithoutSpaces(fieldString);
         }
-        if (element.contains('LastName')) {
-          parsedContact.LastName = element.split(':')[1].replaceAll(' ', '');
+
+        else if (fieldString.contains('LastName') && StaticTask.isRightPartOfPairBlank(fieldString)) {
+          record.LastName = StaticTask.getRightPartOfPairWithoutSpaces(fieldString);
         }
-        if (element.contains('Email')) {
-          if (String.isNotBlank(element.split(':')[1])) {
-            parsedContact.Email = element.split(':')[1].replaceAll(' ', '');
-            emailToContactMap.put(
-              element.split(':')[1].replaceAll(' ', ''),
-              parsedContact
-            );
-          }
+
+        else if (fieldString.contains('Email') && StaticTask.isRightPartOfPairBlank(fieldString)) {
+          record.Email = StaticTask.getRightPartOfPairWithoutSpaces(fieldString);
         }
       }
+
+      contactList.add(record);
+    }
+
+    return contactList;
+  }
+
+
+  public static Map<String, Contact> createEmailAndContactMap (List<Contact> contactList) {
+
+    Map<String, Contact> emailAndContactMap = new Map<String, Contact>();
+
+    for (Contact record : contactList) {
+      if ((String) record.get('Email') != null) {
+        emailAndContactMap.put((String) record.get('Email'), record);
+      }
     }
-    return emailToContactMap;
+
+    return emailAndContactMap;
+  }
+
+
+  public static String getRightPartOfPairWithoutSpaces (String pair) {
+    return pair.split(':')[1].replaceAll(' ', '');
```

Comment:
This is pretty complex statement. Can you make it more robust and easier to read?

The problem may occur is con.split(':') will return only one result for some reason. You don't need to handle this case, but you need to make sure it's easy to trace

Also, why do you need to replace spaces? Is there any better way to do this rather then replacing them?